### PR TITLE
Statistics test: check if distributions match torch

### PR DESCRIPTION
**Pull request by jla524**
To support #756 with the distribution test.

### CODE REVIEW

**Comment by jla524** [comment]
The distributions match, but the values don't. I'm marking this as a draft for now.
Below is a comparison of `Tensor.rand` and `torch.rand` with `np.testing.assert_allclose`:
```
======================================================================                                                                                                               
FAIL: test_rand (__main__.TestRandomness.test_rand)                                                                                                                                  
----------------------------------------------------------------------                                                                                                               
Traceback (most recent call last):                                                                                                                                                   
  File "/Users/jacky/fun/tinygrad/test/test_randomness.py", line 58, in test_rand                                                                                                    
    self.assertTrue(equal_distribution(Tensor.rand, torch.rand, lambda x: np.random.rand(*x)))                                                                                       
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                        
  File "/Users/jacky/fun/tinygrad/test/test_randomness.py", line 53, in equal_distribution                                                                                           
    return np.testing.assert_allclose(x, y) and kstest(x, z) >= alpha                                                                                                                
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                          
  File "/opt/homebrew/lib/python3.11/site-packages/numpy/testing/_private/utils.py", line 1592, in assert_allclose                                                                   
    assert_array_compare(compare, actual, desired, err_msg=str(err_msg),                                                                                                             
  File "/opt/homebrew/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/contextlib.py", line 81, in inner                                         
    return func(*args, **kwds)                                                                                                                                                       
           ^^^^^^^^^^^^^^^^^^^                                                                                                                                                       
  File "/opt/homebrew/lib/python3.11/site-packages/numpy/testing/_private/utils.py", line 862, in assert_array_compare                                                               
    raise AssertionError(msg)                                                                                                                                                        
AssertionError:                                                                                                                                                                      
Not equal to tolerance rtol=1e-07, atol=0                                                                                                                                            
                                                                                                                                                                                     
Mismatched elements: 460 / 460 (100%)                                                                                                                                                
Max absolute difference: 0.94651692                                                                                                                                                  
Max relative difference: 314.62465932                                                                                                                                                
 x: array([0.546324, 0.878102, 0.727007, 0.185528, 0.540205, 0.9209  ,                                                                                                               
       0.391524, 0.946566, 0.289505, 0.874508, 0.461689, 0.115743,                                                                                                                   
       0.059763, 0.193732, 0.165006, 0.341737, 0.276561, 0.495791,...                                                                                                                
 y: array([0.262025, 0.158684, 0.278127, 0.459317, 0.321001, 0.518393,                                                                                                               
       0.261943, 0.976085, 0.732815, 0.115274, 0.386275, 0.628501,                                                                                                                   
       0.125058, 0.983549, 0.443225, 0.789558, 0.794119, 0.361262,...                                                                                                                
```

**Comment by jla524** [comment]
I couldn't get the values to match (i.e. `np.testing.assert_allclose`), but the distributions are identical.

File: test/test_randomness.py [hunk 1]
```diff
@@ -37,39 +38,40 @@ def kstest(l1, l2):
...
+  x = func(*shape).cpu().numpy().flatten()
+  y = np.random.randn(*shape).flatten()
+  return kstest(x, y) >= alpha
+
+def equal_distribution(tiny_func, torch_func, numpy_func, shape=(20, 23), alpha=0.05):
   Tensor.manual_seed(1337)
+  torch.manual_seed(1337)
   np.random.seed(1337)
-  x = tinygrad_func(*shape).cpu().numpy().flatten()
+  x = tiny_func(*shape).cpu().numpy().flatten()
```
Comment (geohot) [inline test/test_randomness.py:50]: Should torch_func be called in here?

**Comment by geohot** [comment]
Sweet!

**Pull request merged**